File: 1_algorithmic_toolbox/week2/cpp/7_gcd.cpp
#include <iostream>

int gcd(int a, int b)
{
  while (b) {
    auto t { a % b };
    a = b;
    b = t;
  }
  return a;
}

int main()
{
  int a {}, b {};
  std::cin >> a >> b;
  std::cout << gcd(a, b) << '\n';
}

File: 1_algorithmic_toolbox/week2/rust/7_gcd.rs
fn main() {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    let mut input = input.split_whitespace().map(|i| i.parse::<i32>().unwrap());

    let a = input.next().unwrap();
    let b = input.next().unwrap();

    println!("{}", gcd(a, b));
}

fn gcd(mut a: i32, mut b: i32) -> i32 {
    while b > 0 {
        (a, b) = (b, a % b);
    }
    a
}
